orchagent/intfsorch.cpp

```diff
@@ -226,6 +227,81 @@ bool IntfsOrch::setRouterIntfsAdminStatus(const Port &port)
     return true;
 }
 
+bool IntfsOrch::setIntfVlanFloodType(const Port &port)
+{
+    SWSS_LOG_ENTER();
+
+    if (port.m_type != Port::VLAN)
+    {
+        SWSS_LOG_ERROR("VLAN flood type cannot be set for non VLAN interface \"%s\"", port.m_alias.c_str());
+        return false;
+    }
+
+    sai_attribute_t attr;
+    attr.id = SAI_VLAN_ATTR_BROADCAST_FLOOD_CONTROL_TYPE;
+    attr.value.s32 = port.m_vlan_info.vlan_flood_type;
+
+    sai_status_t status = sai_vlan_api->set_vlan_attribute(port.m_vlan_info.vlan_oid, &attr);
+    if (status != SAI_STATUS_SUCCESS)
+    {
+        SWSS_LOG_ERROR("Failed to set flood type for VLAN %u, rv:%d", port.m_vlan_info.vlan_id, status);
+        return false;
+    }
+
+    return true;
+}
+
+bool IntfsOrch::setIntfProxyArp(const string &alias, const string &proxy_arp)
+{
+    SWSS_LOG_ENTER();
+
+    if (proxy_arp != "enabled" || proxy_arp != "disabled")
+    {
+        SWSS_LOG_ERROR("Proxy ARP state is invalid: \"%s\"", proxy_arp.c_str());
+        return false;
+    }
+
+    if (m_syncdIntfses.find(alias) == m_syncdIntfses.end())
+    {
+        SWSS_LOG_ERROR("Interface \"%s\" doesn't exist", alias.c_str());
+        return false;
+    }
+
+    if (m_syncdIntfses[alias].proxy_arp == (proxy_arp == "enabled" ? true : false))
+    {
+        SWSS_LOG_INFO("Proxy ARP is already set to \"%s\" on interface \"%s\"", proxy_arp.c_str(), alias.c_str());
+        return true;
+    }
+
+    Port port;
+    if (!gPortsOrch->getPort(alias, port))
+    {
+        SWSS_LOG_ERROR("Failed to get port info for the interface \"%s\"", alias.c_str());
+        return false;
+    }
+
+    if (port.m_type == Port::VLAN)
+    {
+        if (proxy_arp == "enabled")
+        {
+            port.m_vlan_info.vlan_flood_type = SAI_VLAN_FLOOD_CONTROL_TYPE_NONE;
+        }
+        else
+        {
+            port.m_vlan_info.vlan_flood_type = SAI_VLAN_FLOOD_CONTROL_TYPE_ALL;
```

Comment:
I don't see this updated back to port map. Please refer [this ](https://github.com/Azure/sonic-swss/blob/master/orchagent/portsorch.cpp#L2021) after updating a field of port. In this case, i dont think you need to store this value. Just pass the `flood_type` to `setIntfVlanFloodType`

Reply:
Hi @volodymyrsamotiy @prsunny, is it alright to influence the flooding behavior of all BUM packets when proxy_arp enabled? This flood_type attr seems not only applied to ARP packets.
````
typedef enum _sai_vlan_flood_control_type_t
{
    /**
     * @brief Flood on all vlan members
     *
     * When setting all to broadcast or unknown multicast flood, it also includes
     * flooding to the router. When setting all to unknown unicast flood, it does
     * not include flooding to the router
     */
    SAI_VLAN_FLOOD_CONTROL_TYPE_ALL,

    /** Disable flooding */
    SAI_VLAN_FLOOD_CONTROL_TYPE_NONE,

    /** Flood on the L2MC group */
    SAI_VLAN_FLOOD_CONTROL_TYPE_L2MC_GROUP,

    /**
     * @brief Flood on all vlan members and L2MC group
     *
     * Flood on all vlan members, without the router
     * In addition, flood on the supplied L2MC group
     */
    SAI_VLAN_FLOOD_CONTROL_TYPE_COMBINED

} sai_vlan_flood_control_type_t;
````